bin/BuildPackages.sh

```diff
@@ -52,86 +36,203 @@ build_packages() {
 # Even if it doesn't work completely automatically for you, you may get
 # an idea what to do for a complete installation of GAP.
 
+set -e
 
 # Is someone trying to run us from inside the 'bin' directory?
+run_from_bin(){
 if [[ -f gapicon.bmp ]]
 then
   error "This script must be run from inside the pkg directory" \
         "Type: cd ../pkg; ../bin/BuildPackages.sh"
 fi
+}
+
+# set default parameters
+set_default_parameters() {
 
 # CURDIR
 CURDIR="$(pwd)"
 
-# GAPDIR
-GAPDIR=""
-if [[ "$#" -ge 1 ]]
+# default GAPDIR
+pushd .. >/dev/null
+GAPDIR="$(pwd)"
+popd >/dev/null
+
+# default COLOR
+if [[ -t 1 ]]
 then
+  COLOR=1
+else
+  COLOR=0
+fi
+
+# default LOGDIR
+LOGDIR=log
+
+# default LOGFILE
+LOGFILE=buildpackages
+
+# default FAILPKGFILE
+FAILPKGFILE=fail
+}
+
+# collect arguments
+collect_arguments(){
+while [[ "$#" -ge 1 ]]
+do
   case "$1" in
     --with-gaproot=*)
       GAPDIR=$(sed 's|--with-gaproot=||' <<< "$1")
       shift
     ;;
 
+    --with-gaproot)
+      shift
+      GAPDIR="$1"
```

Comment:
Then you can nicely fit each option into a single line (as I do in PR #1128), which makes it much easier to read the whole code block at once -- it just fills a few lines, not a whole page.